mongodbatlas/global_clusters.go

```diff
@@ -0,0 +1,165 @@
+}
+
+//AddManagedNamespace adds a managed namespace to the specified Global Cluster.
+//See more: https://docs.atlas.mongodb.com/reference/api/database-users-create-a-user/
+func (s *GlobalClustersServiceOp) AddManagedNamespace(ctx context.Context, groupID string, clusterName string, createRequest *ManagedNamespace) (*GlobalCluster, *Response, error) {
+	if createRequest == nil {
+		return nil, nil, NewArgError("createRequest", "cannot be nil")
+	}
+
+	path := fmt.Sprintf(globalClustersBasePath, groupID, clusterName, "managedNamespaces")
+
+	req, err := s.client.NewRequest(ctx, http.MethodPost, path, createRequest)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(GlobalCluster)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+//DeleteManagedNamespace deletes the managed namespace configuration of the global cluster given.
+//See more: https://docs.atlas.mongodb.com/reference/api/global-clusters-delete-namespace/
+func (s *GlobalClustersServiceOp) DeleteManagedNamespace(ctx context.Context, groupID string, clusterName string, deleteRequest *ManagedNamespace) (*GlobalCluster, *Response, error) {
+	if deleteRequest == nil {
+		return nil, nil, NewArgError("createRequest", "cannot be nil")
+	}
+
+	path := fmt.Sprintf(globalClustersBasePath, groupID, clusterName, "managedNamespaces")
+
+	req, err := s.client.NewRequest(ctx, http.MethodDelete, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	q := req.URL.Query()
+	q.Add("collection", deleteRequest.Collection)
+	q.Add("db", deleteRequest.Db)
+	req.URL.RawQuery = q.Encode()
+
+	root := new(GlobalCluster)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+//AddCustomZoneMappings adds an entry to the list of custom zone mappings for the specified Global Cluster.
+//See more: https://docs.atlas.mongodb.com/reference/api/global-clusters-add-customzonemapping/
+func (s *GlobalClustersServiceOp) AddCustomZoneMappings(ctx context.Context, groupID string, clusterName string, createRequest *CustomZoneMappingsRequest) (*GlobalCluster, *Response, error) {
+	if createRequest == nil {
+		return nil, nil, NewArgError("createRequest", "cannot be nil")
+	}
+
+	path := fmt.Sprintf(globalClustersBasePath, groupID, clusterName, "customZoneMapping")
+
+	req, err := s.client.NewRequest(ctx, http.MethodPost, path, createRequest)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(GlobalCluster)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+//DeleteManagedNamespace removes all custom zone mappings from the specified Global Cluster.
+//See more: https://docs.atlas.mongodb.com/reference/api/global-clusters-delete-namespace/
+func (s *GlobalClustersServiceOp) DeleteCustomZoneMappings(ctx context.Context, groupID string, clusterName string) (*GlobalCluster, *Response, error) {
```

Comment:
The comment (DeleteManagedNamespace) doesn't have a relation with the function name (DeleteCustomZoneMappings)